charts/kong/templates/_helpers.tpl

```diff
@@ -761,8 +763,14 @@ Environment variables are sorted alphabetically
 - name: wait-for-postgres
 {{- if .Values.waitImage.unifiedRepoTag }}
   image: "{{ .Values.waitImage.unifiedRepoTag }}"
-{{- else }}
+{{- else if .Values.waitImage.repository }}
   image: "{{ .Values.waitImage.repository }}:{{ .Values.waitImage.tag }}"
+{{- else }} {{/* default to the Kong image */}}
+{{- if .Values.image.unifiedRepoTag }}
+  image: "{{ .Values.image.unifiedRepoTag }}"
+{{- else }}
+  image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+{{- end }}
```

Comment:
This `if` ladder seems to be adding some duplicated and rather complex logic. Could this be avoided by something similar to the below:

- implementing a helper template (called e.g. `getRepoTag`) that takes an "image" object as scope, and returns either `.unifiedRepoTag` or a `{{.repository}}:{{.tag}}`
- using the `default` function, something like this: `image: {{ default (include "getRepoTag" .Values.Image) (include "getRepoTag" .Values.Image) }}`

?

Reply:
Yeah, good idea. This gets repeated in several locations and doesn't need to be. The tests for it were also incorrect. 3f64601